### DISCUSSION TITLE
Question about solution of the equation

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#21833] [Q&A General] [answered]
There are 3 equations(a,b,c) in kernel. Can I write the discretization of another equation(d) in userobject and solve these four equations together? Are there any examples I can look at?

**Comment by GiudGiud** [comment, marked as answer]
Hello

The number of equations is dictated by the number of variables.
You may form 3 equations with kernels, and write another one with a user object. 

There are numerous examples of user objects contributing to the residual in:
the tensor mechanics module
https://github.com/idaholab/moose/tree/next/modules/tensor_mechanics/src/userobjects/lagrangian
the thermal hydraulics module
https://github.com/idaholab/moose/blob/next/modules/thermal_hydraulics/src/userobjects/ADBoundaryFlux3EqnFreeOutflow.C
the navier stokes module
https://github.com/idaholab/moose/blob/next/modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C
the ray tracing module as well (ray kernels) though there s a lot of inheritance going on, from a UO to a ray-kernel.

Guillaume
